File: Library-Service/DTOs/LoanDtos.cs
using System.ComponentModel.DataAnnotations;
using Library_Service.Models;

namespace Library_Service.DTOs;

public record LoanGetDto(
    Guid Id,
    string BorrowerName,
    DateTime LoanDate,
    DateTime? ReturnDate,
    Status Status,
    DateTime CreatedAt,
    DateTime? UpdatedAt,
    string BookTitle
);

public record LoanDetailGetDto(
    Guid Id,
    string BorrowerName,
    DateTime LoanDate,
    DateTime? ReturnDate,
    Status Status,
    DateTime CreatedAt,
    DateTime? UpdatedAt,
    BookSummaryDto Book
);

public record LoanCreateDto(
    [Required(ErrorMessage = "O nome do locatário é obrigatório.")]
    [StringLength(150, MinimumLength = 3, ErrorMessage = "O nome do locatário deve ter entre 3 e 150 caracteres.")]
    string BorrowerName,
    [Required(ErrorMessage = "O ID do livro é obrigatório.")]
    Guid BookId
);

File: Library-Frontend/src/types/loanDtos.ts
import type { BookSummaryDto } from "./commonDtos";

export type LoanStatus = 1 | 2;

export const getLoanStatusLabel = (status: LoanStatus): string => {
  return status === 1 ? "Ativo" : "Devolvido";
};

export type LoanGetDto = {
  id: string;
  borrowerName: string;
  loanDate: string;
  returnDate: string | null;
  status: LoanStatus;
  createdAt: string;
  updatedAt: string | null;
  bookTitle: string;
};

export type LoanDetailGetDto = {
  id: string;
  borrowerName: string;
  loanDate: string;
  returnDate: string | null;
  status: LoanStatus;
  createdAt: string;
  updatedAt: string | null;
  book: BookSummaryDto;
};

export type LoanCreateDto = {
  borrowerName: string;
  bookId: string;
};
